docs/source/ProgrammingGuide/DeprecatedInitialization.md

```diff
@@ -0,0 +1,117 @@
+
+Command-line arguments come in "prefixed" and "non-prefixed" versions. Prefixed versions start with the string `--kokkos-`. [`Kokkos::initialize`](../API/core/initialize_finalize/initialize) will remove prefixed options from the input list, but will preserve non-prefixed options. Argument options are given with an equals (`=`) sign. If the same argument occurs more than once, the last one is used. Furthermore, prefixed versions of the command line arguments take precedence over the non-prefixed ones. For example, the arguments
+
+    --kokkos-threads=4 --threads=2
+
+set the number of threads to 4, while
+
+    --kokkos-threads=4 --threads=2 --kokkos-threads=3
+
+set the number of threads to 3. Table 5.1 gives a full list of command-line options.
+
+<h4>Table 15.1: Command-line options for Kokkos::initialize</h4>
+
+Argument | Description
+:---      | :---
+--kokkos-help  <br/> --help   | print this message
+--kokkos-threads=INT <br/> --threads=INT  | specify total number of threads or number of threads per NUMA region if used in conjunction with `--numa` option.
+--kokkos-numa=INT <br/> --numa=INT | specify number of NUMA regions used by process.
+--device-id=INT | specify device id to be used by Kokkos.
+--num-devices=INT[,INT] | used when running MPI jobs. Specify the number of devices per node to be used. Process to device mapping happens by obtaining the local MPI rank and assigning devices round-robin. The optional second argument allows for an existing device to be ignored. This is most useful on workstations with multiple GPUs, one of which is used to drive screen output.
+
+***
+<sup>1</sup> This is the preferred set of defaults when CUDA and OpenMP are enabled. If you use a thread-parallel host execution space, we prefer Kokkos' OpenMP back-end, as this ensures compatibility of Kokkos' threads with the application's direct use of OpenMP threads. Kokkos cannot promise that its Threads back-end will not conflict with the application's direct use of operating system threads.
+
+## 15.2 Initialization by struct
+
+Instead of giving [`Kokkos::initialize()`](../API/core/initialize_finalize/initialize) command-line arguments, one may directly pass in initialization parameters using the following struct:
+
+```c++
+struct Kokkos::InitArguments {
+  int num_threads;
+  int num_numa;
+  int device_id;
+  int ndevices;
+  int skip_device;
+  bool disable_warnings;
+};
+```
+The `num_threads` field corresponds to the `--kokkos-threads` command-line argument, `num_numa` to `--kokkos-numa`, `device_id` to `--device-id`, `ndevices` to the first value in `--num-devices` and `skip_devices` to the second value in `--num-devices`. (See Table 5.1 for details.) Not all parameters are observed by all execution spaces, and the struct might expand in the future if needed.
+
```

Comment:
Is it important to say which parameters are not used / visible by which execution spaces?